File: lab9/backend_squares/squares/squares_app/serializers/friend.py
from rest_framework import serializers

from .user import UsersSerializer

class FriendsSerializer(serializers.Serializer):
    id = serializers.IntegerField(write_only=True)
    user_friend = UsersSerializer(read_only=True)
    friend = UsersSerializer(read_only=True)
    status = serializers.CharField(max_length=3, required=False)

class FriendsUpdateSerializer(serializers.Serializer):
    ready_to_play = serializers.BooleanField()

File: project/angular_frontend/squares/src/app/models/dto/friend.ts
import {User} from "./user";

export interface Friendship {
  user_friend?: User;
  friend?: User;
  status?: string;
}

export interface Get_friends {
  Friends?: Friendship[];
}

export interface Post_friends {
  Friendship?: {
    id?: number;
    status?: string;
  };
}

export interface Patch_friends_friendid {
  Status?: {
    ready_to_play?: boolean;
  };
}
